src/java/com/puppetlabs/puppetserver/jruby/ScriptingContainer.java

```diff
@@ -0,0 +1,35 @@
+package com.puppetlabs.puppetserver.jruby;
+
+import org.jruby.embed.LocalContextScope;
+import org.jruby.embed.LocalVariableBehavior;
+
+/**
+ * An extension of the JRuby ScriptingContainer class which is
+ * slightly easier to use from Clojure
+ */
+public class ScriptingContainer
+        extends org.jruby.embed.ScriptingContainer {
+    public ScriptingContainer(LocalContextScope scope) {
+        super(scope);
+    }
+    public ScriptingContainer(LocalContextScope scope, LocalVariableBehavior behavior) {
+        super(scope, behavior);
+    }
+
+    /**
+     * This method delegates to a specific signature of #callMethod from the
```

Comment:
Believe it or not, adding a schema to the return value of the `create-scripting-container` causes the error.  When that function is not schematized, or when the schema omits a return value specification, Clojure calls the correct Java signature.  With a schematized return value it calls the wrong signature.